src/bin/gpfdist/regress/input/exttab1.source

```diff
@@ -456,6 +456,52 @@ format 'csv';
 select count(*) from bad_location2;
 drop external table bad_location2;
 
+--
+-- Test DEFAULT and NOT NULL
+--
+
+-- Test NOT NULL for readable (constraint should be ignored)
+CREATE EXTERNAL TABLE readable_not_null (a int, b int NOT NULL, c int NULL)
+LOCATION ('gpfdist://@hostname@:7070/exttab1/data_with_null.data')
+FORMAT 'text' (DELIMITER '|' NULL AS 'null');
+
+SELECT a FROM readable_not_null WHERE b IS NULL; -- should work
+SELECT a FROM readable_not_null WHERE c IS NULL; -- should work
+
+-- Test NOT NULL with FILL MISSING FIELDS (constraint should be ignored)
+CREATE EXTERNAL TABLE readable_not_null_fill (a int, b int NOT NULL, c int NULL)
+LOCATION ('gpfdist://@hostname@:7070/exttab1/missing_fields1.data')
+FORMAT 'text' (DELIMITER '|' FILL MISSING FIELDS);
+
+SELECT a FROM readable_not_null_fill WHERE b IS NULL; -- should work
```

Comment:
what does "work" mean? Successful select? Please clarify.

Reply:
Meaning that it will (correctly) find the row where b is NULL. I'll add a comment for this

Reply:
Suggest s/find the row where b is NULL/find the row despite b is NULL in the query/